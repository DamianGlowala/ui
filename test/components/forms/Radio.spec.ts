import { describe, it, expect } from 'vitest'
import { URadio } from '#components'
import type { TypeOf } from 'zod'
import ComponentRender from '../component-render'

describe('Radio', () => {
  it.each([
    ['basic case', {}],
    ['with custom id', { props: { id: 'custom-radio-id' } }],
    ['with value', { props: { value: 'option1' } }],
    ['with name', { props: { name: 'radioName' } }],
    ['with disabled', { props: { disabled: true } }],
    ['with help', { props: { help: 'Some help text' } }],
    ['with label', { props: { label: 'Radio Label' } }],
    ['with required', { props: { required: true } }],
    ['with custom color', { props: { color: 'blue' } }],
    ['with custom class', { props: { class: 'custom-class' } }]
    // @ts-ignore
  ])('renders %s correctly', async (nameOrHtml: string, options: TypeOf<typeof Radio.props>) => {
    if (options !== undefined) {
      options.slots = options.slots || { default: () => 'label' }
      options.slots.default = options.slots.default || (() => 'label')
    }
    const html = await ComponentRender(nameOrHtml, options, URadio)
    expect(html).toMatchSnapshot()
  })
})
