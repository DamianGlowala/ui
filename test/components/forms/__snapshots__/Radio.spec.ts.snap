// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Radio > renders basic case correctly 1`] = `
"<div class="">
  <div class=""><input id="Ooz6KnLRIo_0" type="radio" class=""></div>
  <!--v-if-->
</div>"
`;

exports[`Radio > renders with custom color correctly 1`] = `
"<div class="">
  <div class=""><input id="Ooz6KnLRIo_8" type="radio" class=""></div>
  <!--v-if-->
</div>"
`;

exports[`Radio > renders with custom id correctly 1`] = `
"<div class="">
  <div class=""><input id="custom-radio-id" type="radio" class=""></div>
  <!--v-if-->
</div>"
`;

exports[`Radio > renders with disabled correctly 1`] = `
"<div class="">
  <div class=""><input id="Ooz6KnLRIo_3" disabled="" type="radio" class=""></div>
  <!--v-if-->
</div>"
`;

exports[`Radio > renders with help correctly 1`] = `
"<div class="">
  <div class=""><input id="Ooz6KnLRIo_6" type="radio" class=""></div>
  <div class=""><label for="Ooz6KnLRIo_6" class="">Radio Label
      <!--v-if-->
    </label>
    <p class="">Help text</p>
  </div>
</div>"
`;

exports[`Radio > renders with inputClass correctly 1`] = `
"<div class="">
  <div class=""><input id="Ooz6KnLRIo_9" type="radio" class=""></div>
  <!--v-if-->
</div>"
`;

exports[`Radio > renders with label correctly 1`] = `
"<div class="">
  <div class=""><input id="Ooz6KnLRIo_4" type="radio" class=""></div>
  <div class=""><label for="Ooz6KnLRIo_4" class="">Radio Label
      <!--v-if-->
    </label>
    <!--v-if-->
  </div>
</div>"
`;

exports[`Radio > renders with label slot correctly 1`] = `
"<div class="">
  <div class=""><input id="Ooz6KnLRIo_5" type="radio" class=""></div>
  <div class=""><label for="Ooz6KnLRIo_5" class="">label slot
      <!--v-if-->
    </label>
    <!--v-if-->
  </div>
</div>"
`;

exports[`Radio > renders with name correctly 1`] = `
"<div class="">
  <div class=""><input id="Ooz6KnLRIo_2" name="radioName" type="radio" class=""></div>
  <!--v-if-->
</div>"
`;

exports[`Radio > renders with required correctly 1`] = `
"<div class="">
  <div class=""><input id="Ooz6KnLRIo_7" required="" type="radio" class=""></div>
  <div class=""><label for="Ooz6KnLRIo_7" class="">Radio Label<span class="">*</span></label>
    <!--v-if-->
  </div>
</div>"
`;

exports[`Radio > renders with value correctly 1`] = `
"<div class="">
  <div class=""><input id="Ooz6KnLRIo_1" type="radio" class="" value="option1"></div>
  <!--v-if-->
</div>"
`;
